lib/file-list.js

```diff
@@ -95,13 +95,12 @@ class FileList {
               return this._preprocess(file).then(() => file)
             }
           })
-          .then((files) => {
-            this.buckets.set(pattern, files)
 
-            if (_.isEmpty(files)) {
-              log.warn(`All files matched by "${pattern}" were excluded or matched by prior matchers.`)
-            }
-          })
+        this.buckets.set(pattern, files)
+
+        if (_.isEmpty(files)) {
+          log.warn(`All files matched by "${pattern}" were excluded or matched by prior matchers.`)
+        }
       })
       .then(() => {
```

Comment:
The `then` resolution function is confusing and which promise chain we are resolving is confusing: can we find a better way? At least removing a nesting level might help.

Reply:
Unfortunately I don't have now idea how to rewrite it